phpunit/class-rest-styles-controller-test.php

```diff
@@ -0,0 +1,287 @@
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertEquals( 'view', $data['endpoints'][0]['args']['context']['default'] );
+		$this->assertEquals( array( 'view', 'embed', 'edit' ), $data['endpoints'][0]['args']['context']['enum'] );
+	}
+
+	/**
+	 * Test multiple styles.
+	 */
+	public function test_get_items() {
+		wp_set_current_user( self::$admin_id );
+		$request = new WP_REST_Request( 'GET', '/__experimental/styles' );
+		$request->set_query_params( array( 'dependency' => 'style1' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertCount( 0, $data );
+	}
+
+	/**
+	 * Test multiple styles with nested dependencies.
+	 */
+	public function test_get_items_nested_deps1() {
+		wp_set_current_user( self::$admin_id );
+		$request = new WP_REST_Request( 'GET', '/__experimental/styles' );
+		$request->set_query_params( array( 'dependency' => 'style-with-deps' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertCount( 2, $data );
+
+		$this->assertEquals( 'dependency1', $data[0]['handle'] );
+		$this->assertEquals( 'dependency2', $data[1]['handle'] );
+
+		$this->assertCount( 0, $data[0]['deps'] );
+		$this->assertCount( 0, $data[1]['deps'] );
+	}
+
+	/**
+	 * Test multiple styles with nested dependencies.
+	 */
+	public function test_get_items_nested_deps2() {
+		wp_set_current_user( self::$admin_id );
+		$request = new WP_REST_Request( 'GET', '/__experimental/styles' );
+		$request->set_query_params( array( 'dependency' => 'style-with-nested-deps' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertCount( 3, $data );
+
+		$this->assertEquals( 'dependency3', $data[0]['handle'] );
+		$this->assertEquals( 'dependency4', $data[1]['handle'] );
+		$this->assertEquals( 'dependency5', $data[2]['handle'] );
+
+		$keys = array( 'src', 'url', 'args', 'ver', 'extra', 'textdomain', 'translations_path', 'deps', '_links' );
+		for ( $i = 0; $i < 3; $i ++ ) {
+			foreach ( $keys as $key ) {
+				$this->assertArrayHasKey( $key, $data[ $i ] );
+			}
+		}
+
+		$this->assertCount( 0, $data[0]['deps'] );
+		$this->assertCount( 1, $data[1]['deps'] );
+		$this->assertEquals( array( 'dependency3' ), $data[1]['deps'] );
+		$this->assertCount( 1, $data[2]['deps'] );
+		$this->assertEquals( array( 'dependency4' ), $data[2]['deps'] );
+	}
+
+	/**
+	 * Test single style.
+	 */
+	public function test_get_item() {
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', '/__experimental/styles/' . self::$style_handle );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( self::$style_handle, $data['handle'] );
+		$this->assertEquals( home_url( '/test.css' ), $data['src'] );
+		$this->assertEquals( home_url( '/test.css' ), $data['url'] );
+	}
+
```

Comment:
Add tests for no user / subscriber / author to see that permissions checks work.

Reply:
Added